### PR TITLE
V1 CreateService and StartJobUpdate Timeout signal

### PR DESCRIPTION
**Pull request by rdelval**
* CreateService and StartJobUpdate now stop retries when they encounter a timeout and return an error that satisfies a Timeout interface back to the user. Users may use `realis.IsTimeout()` to determine if the error was a time out error and react accordingly to it.
* Added test that reflects this corner case.
* Code has been refactored to move admin functions to a new file.
* Error messages have been changed to have a lower case first character.
* Removing dead code for initialization of transport protocols.

### CODE REVIEW

File: realis_e2e_test.go [hunk 1]
```diff
@@ -461,6 +465,37 @@ func TestRealisClient_CreateService(t *testing.T) {
...
+	// Test case where http connection timeouts out.
+	t.Run("TimeoutError", func(t *testing.T) {
+		timeoutClient, err := realis.NewRealisClient(realis.SchedulerUrl(AuroraURL),
+			realis.BasicAuth("aurora", "secret"),
+			realis.TimeoutMS(20))
+		assert.NoError(t, err)
+
+		defer timeoutClient.Close()
+
+		// Make sure a timedout error was returned
```
Comment (at15) [inline realis_e2e_test.go]: Maybe you should move/copy this logic to example so people know how to handle timeout properly (copy & paste) when doing update timeout.

Also I didn't see test case/example where an invalid job update setting is given, maybe it's already there thus didn't show up in diff view.

File: retry.go [hunk 2]
```diff
@@ -169,7 +170,20 @@ func (r *realisClient) thriftCallWithRetries(thriftCall auroraThriftCall) (*auro
...
+						return nil, errors.Wrap(clientErr, "permanent connection error")
+					}
+
+					// Corner case where thrift payload was received by Aurora but connection timedout before Aurora was
+					// able to reply. In this case we will return whatever response was received and a TimedOut behaving
+					// error. Users can take special action on a timeout by using IsTimedout and reacting accordingly.
+					if e.Timeout() {
+						timeouts++
+						r.logger.DebugPrintf(
+							"Client closed connection (timedout) %v times before server responded, consider increasing connection timeout",
```
Comment (at15) [inline retry.go]: why not use `%d`

File: retry.go [hunk 3]
```diff
@@ -183,7 +197,7 @@ func (r *realisClient) thriftCallWithRetries(thriftCall auroraThriftCall) (*auro
 			// If there was no client error, but the response is nil, something went wrong.
 			// Ideally, we'll never encounter this but we're placing a safeguard here.
 			if resp == nil {
-				return nil, errors.New("Response from aurora is nil")
+				return nil, errors.New("response from aurora is nil")
```
Comment (at15) [inline retry.go]: Could define a sentinel error `realis.ErrNilResponse` like `io.EOF`, though consider x/erros is comming in go1.13, you might want to defer error handling stuff until that. (btw: there are still people using go1.9 w/ gorealis)

File: realis.go [hunk 4]
```diff
@@ -542,13 +503,15 @@ func (r *realisClient) GetInstanceIds(key *aurora.JobKey, states []aurora.Schedu
 
 	r.logger.DebugPrintf("GetTasksWithoutConfigs Thrift Payload: %+v\n", taskQ)
 
-	resp, retryErr := r.thriftCallWithRetries(func() (*aurora.Response, error) {
-		return r.client.GetTasksWithoutConfigs(nil, taskQ)
-	})
+	resp, retryErr := r.thriftCallWithRetries(
```
Comment (at15) [inline realis.go]: This retry and retryErr is almost same in all the func ... though I am not sure if there are better ways because all of them have a few difference in message and arguments

File: realis_e2e_test.go [hunk 5]
```diff
@@ -35,11 +35,13 @@ var r realis.Realis
 var monitor *realis.Monitor
 var thermosPayload []byte
 
+const AuroraURL = "http://192.168.33.7:8081"
```
[the remark this answers is not in the document]
Comment (rdelval) [reply realis_e2e_test.go]: Good catch!

File: realis_e2e_test.go (hunk 1 above)
Comment (rdelval) [reply realis_e2e_test.go]: Invalid payload + timeout should definitely be tested, I'll add that test today. I'll add to the docs how to handle a timeout.

File: retry.go (hunk 2 above)
Comment (rdelval) [reply retry.go]: Laziness 😄

File: realis.go (hunk 4 above)
Comment (rdelval) [reply realis.go]: Yeah it's tough to make this match all functions. This should have really been like what it's like in V2 all along.

File: realis_admin.go [hunk 6]
```diff
@@ -0,0 +1,269 @@
+package realis
+
+import (
+	"github.com/paypal/gorealis/gen-go/apache/aurora"
+	"github.com/pkg/errors"
+)
+
+// TODO(rdelvalle): Consider moving these functions to another interface. It would be a backwards incompatible change,
```
[the remark this answers is not in the document]
Comment (rdelval) [reply realis_admin.go]: This is where it gets complicated. Server has no separate user/password as far as I can tell. It opens up the door for anyone with username, password to trigger admin functions. Moving all admin functionality to another receiver type would make it that much harder to accidentally fire off an admin call.
